sparkmagic/sparkmagic/livyclientlib/livysession.py

```diff
@@ -110,11 +110,22 @@ def __str__(self):
         return u"Session id: {}\tYARN id: {}\tKind: {}\tState: {}\n\tSpark UI: {}\n\tDriver Log: {}"\
             .format(self.id, self.get_app_id(), self.kind, self.status, self.get_spark_ui_url(), self.get_driver_log_url())
 
+    def _translate_to_livy_kind(self, properties, kind):
+        # Livy does not support "pyspark3" as the kind from 0.4 onwards and only "pyspark" is valid.
+        # That's why we are leveraging PYSPARK_PYTHON_PARAM here to specifyf different python versions.
+        if kind == constants.SESSION_KIND_PYSPARK:
+            python_properties = {constants.PYSPARK_PYTHON_PARAM: constants.LANG_PYTHON}
+            properties[constants.LIVY_CONF_PARAM] = python_properties
+        elif kind == constants.SESSION_KIND_PYSPARK3:
+            properties[constants.LIVY_KIND_PARAM] = constants.SESSION_KIND_PYSPARK
+            python_properties = {constants.PYSPARK_PYTHON_PARAM: constants.LANG_PYTHON3}
```

Comment:
The python3 path should be configurable, and not a constant.

Reply:
Also this kind of replacement of properties breaks configuration JSON, e.g. below is before and after. Basically it lost 'spark.jars.packages' I had passed over in the conf. 


before 
```
 properties: {'driverMemory': '1G', 'executorCores': 4, 'kind': 'pyspark3', 'conf': {'spark.pyspark.python': 'python3', 'spark.jars.packages': 'com.databricks:spark-avro_2.11:4.0.0'}, 'heartbeatTimeoutInSecond': 120}
```
after
```
 properties: {'driverMemory': '1G', 'executorCores': 4, 'kind': 'pyspark', 'conf': {'spark.pyspark.python': 'python3'}, 'heartbeatTimeoutInSecond': 120}
```